client/src/modules/depots/modals/depot.modal.html

```diff
@@ -22,16 +22,81 @@
       </div>
     </div>
 
-    <div class="checkbox">
-      <label>
-        <input
-          type="checkbox"
-          name="is_warehouse"
-          ng-true-value="1"
-          ng-false-value="0"
-          ng-model="DepotModalCtrl.depot.is_warehouse">
-        <span translate>DEPOT.WAREHOUSE</span>
-      </label>
+    <div class="space-between">
+      <!-- stock entry permissions -->
+      <fieldset>
+        <label translate>STOCK.ENTRY</label>
+        <div class="checkbox">
+          <label>
+            <input type="checkbox" name="allow_entry_purchase" ng-true-value="1" ng-false-value="0" ng-model="DepotModalCtrl.depot.allow_entry_purchase">
+            <span translate>STOCK.ENTRY_PURCHASE</span>
+          </label>
+        </div>
+        <div class="checkbox">
+          <label>
+            <input type="checkbox" name="allow_entry_donation" ng-true-value="1" ng-false-value="0" ng-model="DepotModalCtrl.depot.allow_entry_donation">
+            <span translate>STOCK.ENTRY_DONATION</span>
+          </label>
+        </div>
+        <div class="checkbox">
+          <label>
+            <input type="checkbox" name="allow_entry_integration" ng-true-value="1" ng-false-value="0" ng-model="DepotModalCtrl.depot.allow_entry_integration">
+            <span translate>STOCK.ENTRY_INTEGRATION</span>
+          </label>
+        </div>
+        <div class="checkbox">
+          <label>
+            <input type="checkbox" name="allow_entry_transfer" ng-true-value="1" ng-false-value="0" ng-model="DepotModalCtrl.depot.allow_entry_transfer">
+            <span translate>STOCK.ENTRY_TRANSFER</span>
+          </label>
+        </div>
+      </fieldset>
+
+      <!-- stock exit permissions -->
+      <fieldset>
+        <label translate>STOCK.EXIT</label>
+        <div class="checkbox">
+          <label>
+            <input type="checkbox" name="allow_exit_debtor" ng-true-value="1" ng-false-value="0" ng-model="DepotModalCtrl.depot.allow_exit_debtor">
+            <span translate>STOCK.EXIT_INDIVIDIOUS</span>
+          </label>
+        </div>
+        <div class="checkbox">
+          <label>
+            <input type="checkbox" name="allow_exit_service" ng-true-value="1" ng-false-value="0" ng-model="DepotModalCtrl.depot.allow_exit_service">
+            <span translate>STOCK.EXIT_SERVICE</span>
+          </label>
+        </div>
+        <div class="checkbox">
+          <label>
+            <input type="checkbox" name="allow_exit_transfer" ng-true-value="1" ng-false-value="0" ng-model="DepotModalCtrl.depot.allow_exit_transfer">
```

Comment:
If the depot is not a warehouse, should this be shown?

Reply:
I don't think so in the context of bhima (warehouse only can perform transfer to other depots) but the application must be flexible.

Reply:
Alright - I feel like there might be a hidden corner case here.  The code is fine - I'll merge this and we can discuss it.